tests/src/test/scala/org/http4s/CompressionSpec.scala

```diff
@@ -0,0 +1,36 @@
+package org.http4s
+
+import java.io.ByteArrayOutputStream
+import java.util.Locale
+import java.util.zip.{Deflater, GZIPOutputStream}
+
+import cats.effect.IO
+import fs2.{Chunk, Stream}
+import org.http4s.headers.`Accept-Encoding`
+import org.scalacheck.Prop.forAll
+import org.scalacheck.Properties
+
+class CompressionSpec extends Http4sSpec {
+
+  "Compression" should {
+    checkAll(
+      "decode GZIP encoding",
+      new Properties("GZip") {
+        property("middleware decoding == GZIPOutputStream encoding") = forAll {
+          body: String =>
```

Comment:
I think you could do this on binaries instead of Strings.  You're just testing String codecs on top of everything else.